broker/client/client.go

```diff
@@ -712,3 +496,302 @@ func getPeerInfo(peer *ill_db.Peer, symbol string) (string, iso18626.TypeAgencyI
 	}
 	return name, agencyId, address, email
 }
+
+type transactionContext struct {
+	transaction      *ill_db.IllTransaction
+	requester        *ill_db.Peer
+	selectedSupplier *ill_db.LocatedSupplier
+	selectedPeer     *ill_db.Peer
+	event            events.Event
+	err              error
+	errorComment     string
+}
+
+func (c *Iso18626Client) readTransactionContext(ctx extctx.ExtendedContext, event events.Event, supplierMandatory bool) transactionContext {
```

Comment:
@JanisSaldabols remove err from the transactionContext struct and return it as the second return value. Do not use the errorComment but wrap the original error with new  message like `fmt.Errorf("%s: %w", MsgString, err)`